tests/functional/spawn-on-demand/13-trigger-runahead/flow.cylc

```diff
@@ -0,0 +1,23 @@
+# This workflow should behave the same as if foo.1 did not trigger foo.2.
+
+[scheduling]
+   cycling mode = integer
+   initial cycle point = 1
+   final cycle point = 3
+   runahead limit = P1
+   [[graph]]
+      P1 = foo
+[runtime]
+   [[foo]]
+      script = """
+         cylc__job__wait_cylc_message_started
+         if $((CYLC_TASK_CYCLE_POINT == 1)); then
```

Comment:
This syntax is wrong causing the body of this `if` statement to be ignored skipping the test logic:

```
bash: 1: command not found
```

```suggestion
         if ((CYLC_TASK_CYCLE_POINT == 1)); then
```

Reply:
I think this fixes it (plus shellcheck would say put vars in quotes)
```diff
diff --git a/tests/functional/spawn-on-demand/13-trigger-runahead/flow.cylc b/tests/functional/spawn-on-demand/13-trigger-runahead/flow.cylc
index 60e5708e8..8bfd625dd 100644
--- a/tests/functional/spawn-on-demand/13-trigger-runahead/flow.cylc
+++ b/tests/functional/spawn-on-demand/13-trigger-runahead/flow.cylc
@@ -11,13 +11,13 @@
    [[foo]]
       script = """
          cylc__job__wait_cylc_message_started
-         if $((CYLC_TASK_CYCLE_POINT == 1)); then
+         if [[ "$CYLC_TASK_CYCLE_POINT" == 1 ]]; then
 
-            expected = "foo, 1, running, not-held, not-queued, not-runahead
+            expected="foo, 1, running, not-held, not-queued, not-runahead
 foo, 2, waiting, not-held, not-queued, runahead"
-            diff <(cylc dump -t $CYLC_WORKFLOW_NAME) <(echo "$expected")
+            diff <(cylc dump -t "$CYLC_WORKFLOW_NAME") <(echo "$expected")
 
             # Force trigger next instance while it is runahead limited.
-            cylc trigger $CYLC_WORKFLOW_NAME//foo.2
+            cylc trigger "${CYLC_WORKFLOW_NAME}//2/foo"
          fi
       """
```

Reply:
(Fixed, by mining un-pushed changes on the work computer :grimacing: )